thread_pthread.c

```diff
@@ -109,6 +109,8 @@ struct rb_internal_thread_event_hook {
 static rb_internal_thread_event_hook_t *rb_internal_thread_event_hooks = NULL;
 static pthread_rwlock_t rb_internal_thread_event_hooks_rw_lock = PTHREAD_RWLOCK_INITIALIZER;
 
+static rb_serial_t current_fork_gen = 1; /* We can't use GET_VM()->fork_gen */
```

Comment:
So I believe the timeout on macOS was because we can't safely use `GET_VM()` in some of these function.

Having to keep another `fork_gen` counter is a bit unfortunate, but the advantage is that it's fully scoped to `thread_pthread.c`